tests/unit/components/sl-date-picker-test.js

```diff
@@ -367,22 +367,48 @@ test( 'updateDateRange() - when date outside endDate range we show placeholder t
     );
 });
 
-test( 'unregisterEvents() - listens to correct event', function( assert ) {
+test( 'changeDate listener is added and removed from the correct namespace', function( assert ) {
     const component = this.subject();
-    const inputElement = this.$( 'input.date-picker' )[ 0 ];
+    const inputElement = this.$( 'input.date-picker' );
+    const jQueryData = Ember.get( Ember.$, '_data' );
+    const hasDatePickerNamespace = () => {
+        let hasNamespace = false;
+
+        assert.equal(
+            Ember.typeOf(
+                Ember.get( jQueryData( inputElement[0], 'events' ), 'changeDate' )
```

Comment:
Assign both instances of the usage of `jQueryData( inputElement[0], 'events' )` to a `const` and use it within this test.